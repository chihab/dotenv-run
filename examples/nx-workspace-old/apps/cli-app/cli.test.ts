import { execSync } from "child_process";
import { describe, it, expect } from "vitest";

describe("dotenv-run should give the expected output", () => {
  it("empty", () => {
    const actual = execSync(
      "dotenv-run -f .env.app -- bash -c 'echo \"Hello World $API_USERS\"'",
      {
        encoding: "utf8",
      }
    );
    expect(actual).toMatchSnapshot();
  });

  it("dev", () => {
    const actual = execSync(
      "NODE_ENV=dev dotenv-run -f .env.app -- bash -c 'echo \"Hello World $API_USERS\"'",
      {
        encoding: "utf8",
      }
    );
    expect(actual).toMatchSnapshot();
  });

  it("prod", () => {
    const actual = execSync(
      "NODE_ENV=prod dotenv-run -f .env.app -e prod -- bash -c 'echo \"Hello World $API_USERS\"'",
      {
        encoding: "utf8",
      }
    );
    expect(actual).toMatchSnapshot();
  });

  it("multi", () => {
    const actual = execSync(
      "NODE_ENV=prod dotenv-run -e prod -f .secrets,.env.app -- bash -c 'echo \"Hello World $API_USERS\"'",
      {
        encoding: "utf8",
      }
    );
    expect(actual).toMatchSnapshot();
  });
});
