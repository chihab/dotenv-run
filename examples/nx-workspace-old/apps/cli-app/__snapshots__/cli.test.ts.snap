// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dotenv-run should give the expected output > dev 1`] = `
"Hello World https://dotenv-run.dev/api/v1/users
"
`;

exports[`dotenv-run should give the expected output > empty 1`] = `
"Hello World /api/v1/users
"
`;

exports[`dotenv-run should give the expected output > multi 1`] = `
"Hello World https://dotenv-secret-subapp-prod-local-run.app/api/v1/users
"
`;

exports[`dotenv-run should give the expected output > prod 1`] = `
"Hello World https://dotenv-run.app/api/v1/users
"
`;
